File: ITSE-2457-Programming 2 - OOP/Module 6/Employee/src/Employee.java
public class Employee {
    private String employeeID;
    private String firstName;
    private String lastName;
    private Benefits employeeBenefits;
    private Address employeeHomeAddress;

    public Employee(String employeeID, String firstName, String lastName, Benefits employeeBenefits, Address employeeHomeAddress) {
        this.employeeID = employeeID;
        this.firstName = firstName;
        this.lastName = lastName;
        this.employeeBenefits = employeeBenefits;
        this.employeeHomeAddress = employeeHomeAddress;
    }

    public String getEmployeeID() {
        return employeeID;
    }

    public void setEmployeeID(String employeeID) {
        this.employeeID = employeeID;
    }

    public String getFirstName() {
        return firstName;
    }

    public void setFirstName(String firstName) {
        this.firstName = firstName;
    }

    public String getLastName() {
        return lastName;
    }

    public void setLastName(String lastName) {
        this.lastName = lastName;
    }

    public Benefits getEmployeeBenefits() {
        return employeeBenefits;
    }

    public void setEmployeeBenefits(Benefits employeeBenefits) {
        this.employeeBenefits = employeeBenefits;
    }

    public Address getEmployeeHomeAddress() {
        return employeeHomeAddress;
    }

    public void setEmployeeHomeAddress(Address employeeHomeAddress) {
        this.employeeHomeAddress = employeeHomeAddress;
    }

    public void PrintEmployeeInformation() {
        System.out.println("Employee ID: " + employeeID);
        System.out.println("First Name: " + firstName);
        System.out.println("Last Name: " + lastName);
        employeeBenefits.PrintBenefitsInfo();
        employeeHomeAddress.PrintAddressInformation();
    }
}

File: ITSE-2445-Programming 3 - Data Structures - C++/Module 6 Classes and Objects/Lab 1/Employee.cpp
/*
Filename: Employee.cpp
Filepath: ...\ITSE-2445-Programming 3 - Data Structures - C++\Module 6 Classes and Objects\Lab 1\Employee.cpp
Author: Andrew Pettus
Date Created: September 17, 2023
Date Modified:September 17, 2023
Description: Module 6 Lab 1; Employee.cpp
License: NLR
--------------------------------------------------------------------------------------------------------------------------
*/
#include "Employee.h"                   // Include the Employee header file which contains the class declaration
#include <string>                       // Include the string library to use string class
#include <iostream>                     // Include the iostream library to use standard input/output streams, like cout
using namespace std;                    // Use the standard namespace to avoid having to use std:: before standard library names

// empty Employee constructor implementation
Employee::Employee() {
    idNumber = 0;
    firstName = "";
    lastName = "";
    department = "";
    position = "";
}   

// populated Employee constructor implementation
Employee::Employee(string fName, string lName, int idNum, string dept, string pos) {
    firstName = fName;
    lastName = lName;
    idNumber = idNum;
    department = dept;
    position = pos;
}

// setters
void Employee::setFirstName(const string& newFirstName) {
    firstName = newFirstName;
}
void Employee::setLastName(const string& newLastName) {
    lastName = newLastName;
}
void Employee::setIdNumber(int newIdNumber) {
    idNumber = newIdNumber;
}
void Employee::setDepartment(const string& newDepartment) {
    department = newDepartment;
}
void Employee::setPosition(const string& newPosition) {
    position = newPosition;
}

// getters
string Employee::getFirstName() const {
    return firstName;
}
string Employee::getLastName() const {
    return lastName;
}
int Employee::getIdNumber() const {
    return idNumber;
}
string Employee::getDepartment() const {
    return department;
}
string Employee::getPosition() const {
    return position;
}

// displayEmployeeInfo function definition to output Employee info
void Employee::displayEmployeeInfo() const {
    cout << "\nEmployee Information: \n";         
    cout << "First Name: " << firstName << "\n";
    cout << "Last Name: " << lastName << "\n";
    cout << "ID Number: " << idNumber << "\n";
    cout << "Department: " << department << "\n";
    cout << "Position: " << position << "\n";
}
